ats/pkb/pkb.py

```diff
@@ -45,17 +49,82 @@ def process_relations(node: nodes.ASTNode):
                     parents[node.__stmt_id] = parent.__stmt_id
 
             for child in node.children:
+                nonlocal variable_list
+                nodes_stack.append(child)
                 process_relations(child)
 
+                if isinstance(nodes_stack[-1], nodes.VariableNode):
+                    if nodes_stack[-1].parent.condition == nodes_stack[-1]:
+                        uses[nodes_stack[-1].parent.__stmt_id] = [nodes_stack[-1].name]
+                    else:
+                        variable_list.append(nodes_stack[-1].name)
+
+                if isinstance(nodes_stack[-1].parent, nodes.StmtAssignNode):
+                    if nodes_stack[-1].parent.variable == nodes_stack[-1]:
+                        modifies[nodes_stack[-1].parent.__stmt_id] = [
+                            nodes_stack[-1].name
+                        ]
+                    else:
+                        uses[nodes_stack[-1].parent.__stmt_id] = variable_list
+                    variable_list = []
+
+                if isinstance(nodes_stack[-1], nodes.StmtCallNode):
+                    modifies[nodes_stack[-1].__stmt_id] = [nodes_stack[-1]]
+                    uses[nodes_stack[-1].__stmt_id] = [nodes_stack[-1]]
```

Comment:
```suggestion
                    if nodes_stack[-1].name in modifies:
                      raise ...
                    modifies[nodes_stack[-1].name] = []
                    uses[nodes_stack[-1].name] = []
                    modifies[nodes_stack[-1].__stmt_id] = [modifies[nodes_stack[-1].name]]
                    uses[nodes_stack[-1].__stmt_id] = [uses[nodes_stack[-1].name]]
```

Reply:
Trzebaby jakis blad rzucic jak trafilismy na rekurencje, np `RuntimeError('SIMPLE does not support recursion')`